zerver/webhooks/slack/view.py

```diff
@@ -1,60 +1,226 @@
+    elif channels_map_to_topics == VALID_OPTIONS["SHOULD_BE_MAPPED"]:
+        topic_name = f"channel: {channel}"
+        check_send_webhook_message(request, user_profile, topic_name, content)
+    elif channels_map_to_topics == VALID_OPTIONS["SHOULD_NOT_BE_MAPPED"]:
         check_send_webhook_message(
             request,
             user_profile,
             topic_name,
             content,
+            stream=channel,
         )
-    elif channels_map_to_topics == VALID_OPTIONS["SHOULD_BE_MAPPED"]:
-        # If the webhook URL has a user_specified_topic,
-        # then this topic-channel mapping will not be used.
-        topic_name = f"channel: {channel_name}"
-        check_send_webhook_message(
+    else:
+        raise JsonableError(_("Error: channels_map_to_topics parameter other than 0 or 1"))
+
+
+@webhook_view("Slack", notify_bot_owner_on_invalid_json=False)
+@typed_endpoint
+def api_slack_webhook(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    *,
+    channels_map_to_topics: str | None = None,
+) -> HttpResponse:
+    if request.content_type != "application/json":
+        # Handle Slack's legacy Outgoing Webhook Service payload.
+        expected_legacy_variable = ["user_name", "text", "channel_name"]
+        legacy_payload = {}
+        for variable in expected_legacy_variable:
+            if variable in request.POST:
+                legacy_payload[variable] = request.POST[variable]
+            elif variable in request.GET:  # nocoverage
+                legacy_payload[variable] = request.GET[variable]
+            else:
+                raise RequestVariableMissingError(variable)
+
+        text = convert_slack_formatting(legacy_payload["text"])
+        text = replace_links(text)
+        text = get_message_body(text, legacy_payload["user_name"], [])
+        handle_slack_webhook_message(
             request,
             user_profile,
-            topic_name,
-            content,
+            text,
+            legacy_payload["channel_name"],
+            channels_map_to_topics,
         )
-    elif channels_map_to_topics == VALID_OPTIONS["SHOULD_NOT_BE_MAPPED"]:
-        # This channel-channel mapping will be used even if
-        # there is a channel specified in the webhook URL.
+        return json_success(request)
+
+    try:
+        val = request.body.decode(request.encoding or "utf-8")
+    except UnicodeDecodeError:  # nocoverage
+        raise JsonableError(_("Malformed payload"))
+    payload = to_wild_value("payload", val)
+
+    # Handle initial URL verification handshake for Slack Events API.
+    if is_challenge_handshake(payload):
+        challenge = payload.get("challenge").tame(check_string)
         check_send_webhook_message(
             request,
             user_profile,
-            topic_name,
-            content,
-            stream=channel_name,
+            "Integration events",
+            "Successfully verified webhook URL with Slack!",
```

Comment:
This would be worth posting an example screenshot of in Zulip to discuss if we have better wording we should standardize on for this class of thing.

Reply:
Continued the discussion here: [CZO](https://chat.zulip.org/#narrow/channel/127-integrations/topic/Notification.20for.20succesful.20.20integration.20setup.2E)